src/ex1/script.js

```diff
@@ -0,0 +1,381 @@
+      sortTaskList = sortTaskList.concat(TaskList.filter(x => x.type != TYPE_NOTCOMPLETE && x.type != TYPE_DEADLINE_FINISHED).sort((a, b) => { return b.createDateTime.getTime() - a.createDateTime.getTime(); }));
+    }
+  }
+  else {
+    sortTaskList = sortTaskList.concat(TaskList.filter(x => x.type != TYPE_DEADLINE_FINISHED).sort((a, b) => { return b.createDateTime.getTime() - a.createDateTime.getTime(); }));
+  }
+  return sortTaskList;
+}
+
+function CreateNewListItemElementByTaskObject(taskObject, withSort) {
+
+  // Create a new list item when clicking on the "Add" button
+  const li = document.createElement("li");   //create new list item html element  
+  const textElement = document.createTextNode(taskObject.name); // create text html element with user task input name
+  // if user task input name is empty string, show it at alert dialog, else append the LI element to UL.
+
+  li.setAttribute("id", `task${taskObject.id}`);
+  li.setAttribute("taskId", `${taskObject.id}`);
+
+  DisplayDropdown(); //display dropdown
+  VanishEmptyTaskFloatMassage();
+  li.appendChild(textElement); // append text element to list item element
+  document.getElementById("myUL").appendChild(li);
+  // alert("Task Added Successfully");
+  if (taskObject.isAnimated) {
+    AnimateListAdd(li);
+    updateTask(taskObject.id, taskObject.type, taskObject.name, taskObject.datetime, false)
+  }
+  DisplayInvisibleButtons(); // display sort and clear all buttons
+  li.addEventListener("click", ({ target }) => {
+    CheckedTask(target, false); // mark task for complete status
+  });
+  CreateRemoveItemFromList(li);
+  addCalendarIcon(li, taskObject.datetime);
+  if (taskObject.type == TYPE_FINISHED) {
+    CheckedTask(li, true)
+  }
+  else if (taskObject.type == TYPE_DEADLINE_FINISHED) {
+    li.classList.add('markDeadline');
+  }
+
+
+}
+
+function randomIntFromInterval(min, max) { // min and max included 
+  return Math.floor(Math.random() * (max - min + 1) + min)
+}
+
+function CheckedTask(target, withSort) {
+  console.log(target.tagName);
+  if (target.tagName == "INPUT" || target.tagName == "BUTTON") {
+    return;
+  }
+  // Add a "checked" symbol when clicking on a list item
+  if (!target.classList.contains('checked')) {
+    let taskObject = getTask(target.getAttribute('taskId'));
+    updateTask(taskObject.id, TYPE_FINISHED, taskObject.name, taskObject.datetime, false)
+    if (!withSort) {
+      rotateImage();
+    }
+
+  }
+  else {
+    let taskObject = getTask(target.getAttribute('taskId'));
+    updateTask(taskObject.id, TYPE_NOTCOMPLETE, taskObject.name, taskObject.datetime, false)
+  }
+  target.classList.toggle('checked');
+}
+
+function AddTaskByEnter() {
+  document.getElementById("myInput")
+    .addEventListener("keyup", function (event) {
+      event.preventDefault();
+      if (event.keyCode === 13) {
```

Comment:
To make it more readable you can use the Enter string
```suggestion
      if (event.key === "Enter") {
```